### ISSUE TITLE
Documentation needs better explanations

### ISSUE THREAD

**Issue by filiptronicek** [Zulko/moviepy#1192]
I went to one of the examples in the documentation [link](https://zulko.github.io/moviepy/examples/ukulele_concerto.html) and saw this step:

```python
txt_mov = txt_col.set_pos( lambda t: (max(w/30,int(w-0.5*w*t)),
                                  max(5*h/6,int(100*t))) )
```
I didn't understand what was going on there and wanted to do a reverse animation, but the comment says just: "I am *NOT* explaining the formula, understands who can/want.", which I don't find helpful at all.

**Comment by tburrows13** [comment]
Hey, thanks for pointing that out. We're going to focus on revamping the examples before v2.0 is released, and that is definitely something that needs changing.

I'll give an explanation a go here:

The lambda function needs to return a tuple of (x, y) for a time t.
x = `max(w/30, int(w-0.5*w*t)`. At t=0, the second of those expressions is the largest, and it has t with a negative sign, so as t increases, x decreases linearly. This is why the text scrolls to the left in the video. However, it stops scrolling when it reaches x=`w/30`, because at that point, `w/30` > `int(w-0.5*w*t)`, so x is now a constant.

The exact same thing applies to y, expect that y increases with t instead of decreasing, so it starts constant and when `int(100*t)` becomes greater than `5*h/6`, it starts moving down the screen.

Hopefully this helps :)

**Comment by filiptronicek** [comment]
It really did, thank you. Managed to make my x right but am still unsure how to reverse the x animation. Probably haven't given it enought thought, but any ideas?

**Comment by tburrows13** [comment]
Sure. If we for now disregard the stopping when it gets to a certain place, then the current x is defined as `w - 0.5*w*t` which means that it starts at x=`w`, and moves `0.5*w` pixels a second, which means that it traverses the entire width in 2 seconds. In order to move to the right, you want the start to be, say, `0`, and the speed needs to be positive, for example `0.5 * w`. So you set x equal to `0.5*w*t`.

If you want it to stop at a certain point, then you need to emulate the `max` function, but because we have reversed the direction, you want `min` instead. For example `min(0.9*w, 0.5*w*t)` will move to the right until x equals `0.9 * w`.

**Comment by filiptronicek** [comment]
Awesome works as expected, but how would I go about making the animation start from a not visible part of the view - I need the text to go from the left side, but when I try subtracting it from the width, it seems like it is overtaken by the max function.
```python
txt_mov = txt_clip.set_pos( lambda t: ( # animate the text
        max((w-10000),
        int(0.5*w*t)
    ),
        max(1.8*h/6,
        int(100*t)
        )
    ) 
)
```
![image](https://user-images.githubusercontent.com/29888641/81906840-29c49100-95c7-11ea-953f-e4c4eaec6a0f.png)

**Comment by tburrows13** [comment]
So to recap, you have 3 key constants:

1. The first argument to min/max. This defines the coordinate at which the text should become stationary. Let's call this `a`.
2. The constant inside the `int`. This defines the starting position of the movement. Let's call this `b`.
3. The coefficient of `t` inside the `int`. This defines the speed (in pixels/sec) of the movement. Let's call this `c`.

You need to think carefully about `max` vs `min` and how that relates to having a positive coefficient of `t`. In your example, for the x-coordinate, `b` = 0 so it starts against the left side of the screen. If you want it completely off screen to start, then set `b` equal to `-txt_clip.w`.
`c` is positive so it moves to the right. However, you are using `max` with very negative `a`. Which means that the `b + c*t` is always greater than `a`, so the max is useless. You want `a` to equal something like `0.3*w`.

To sum up, here's what I think you meant for your x-component:

```
max((w*0.3), int(-txt_clip.w, 0.5*w*t))
```


---
It seems like you might be better off defining a proper function with if/else statements instead. Then you have complete control over what happens and you don't have to worry about the min/max difference.

Something like this:

```
def get_position(t):
    """Returns an (x, y) coordinate for a given time t"""
    ...

txt_mov = txt_clip.set_position(get_position)
```

Note: `.set_pos` is deprecated and will be removed in v2.0. Use `.set_position instead.

**Comment by filiptronicek** [comment]
Yes, makes sense to move it to a function. The only thing is that the `-txt_clip.w` centers it more than moves it off screen.
![image](https://user-images.githubusercontent.com/29888641/81925726-2dfda800-95e1-11ea-918e-44d95ce10cea.png)

**Comment by tburrows13** [comment]
Oh sorry, it should be `min((w*0.3), int(-txt_clip.w + 0.5*w*t))`, not `max`.

**Comment by filiptronicek** [comment]
Yes, thanks! Worked perfectly

**Comment by tburrows13** [comment]
I'll leave it open until the original example is updated :)

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
